burotel/indico_burotel/tasks.py

```diff
@@ -0,0 +1,189 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2014 - 2021 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+import requests
+from celery.exceptions import Retry
+from celery.schedules import crontab
+from flask_pluginengine import current_plugin
+from requests.exceptions import RequestException, Timeout
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.types import Interval
+
+from indico.core.auth import multipass
+from indico.core.celery import celery
+from indico.core.db import db
+from indico.modules.rb.models.reservations import Reservation, ReservationEditLog, ReservationState
+from indico.modules.rb.models.room_attributes import RoomAttribute, RoomAttributeAssociation
+from indico.modules.rb.models.rooms import Room
+from indico.modules.users.models.users import User
+from indico.util.date_time import now_utc
+
+from indico_burotel.notifications import notify_about_to_cancel, notify_automatic_cancellation
+
+
+def _get_cern_ident(plugin, user):
+    cern_ident_provider = plugin.settings.get('cern_identity_provider')
+    cern_ident = next(
+        (ident for ident in user.identities if ident.provider == cern_ident_provider),
+        None
+    )
```

Comment:
`get_identity(provider)` would be a useful util on the `User` model ;)

Reply:
https://github.com/indico/indico/pull/4951 😉